File: src/HumanPlayer.cpp
/*
 * HumanPlayer.cpp
 *
 *  Created on: 16 בנוב׳ 2017
 *      Author: noam
 */

#include "HumanPlayer.h"
#include "Player.h"
#include "Position.h"
#include "GameFlow.h"
#include "Board.h"
#include "GameLogic.h"
#include <iostream>
#include <vector>
#include <stdlib.h>
using namespace std;


HumanPlayer::HumanPlayer(char name): Player(name) {}

bool HumanPlayer::playTurn(GameLogic logic) const {

	const vector<Position>& moves = logic.calculateMoves(player);  // find the moves of the player
			if (moves.size() != 0) {  // check the player has moves
				cout << endl<< "player: " << player<<" your possible moves are: ";
				for (unsigned int i = 0; i < moves.size(); i++) {
						cout << " ( " << moves[i].getRow() << ", " << moves[i].getColumn() << ") ";
				}
				cout << endl;
				// checking the input and convert it to integers
				int row, column;
				bool flag = true;     //
				do {
					string r ,c;
					cout << "please enter row number: " << endl ;
					getline(cin, r);
					cout  << "please enter column number: " << endl ;
					getline(cin, c);


					row = atoi(r.c_str());
					column = atoi(c.c_str());

					Position p(row,column); // the position the player enter
					for (unsigned int i = 0; i < moves.size(); i++) {  // check that this position is equal to one of his options
						if (p == moves[i]) {
							flag =false;
							break;
						}
					}
					if (flag){
						cout << "wrong input try again!! " << endl;
					}
				} while (flag);

				Position p(row,column);
				logic.updateBoard(p,player);     // end of turn the main board is update
				return true;
			}
			else {
				return false;
			}


}

File: javaFXEx6/HumanPlayer.java
package javaFXEx6;

import java.io.IOException;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Scanner;

/*
 * Noam shimshoviz 203565429 Sarit Zevin 313242588
 */
public class HumanPlayer extends Player{
	private char player;
	public HumanPlayer(char name) {
		super(name);
	}
	public boolean playTurn(GameLogic logic) throws IOException {
		LinkedList<Position> moves = logic.calculateMoves(this.getPlayer());  // find the moves of the player
		if (!moves.isEmpty()) {  // check the player has moves
			System.out.println("");
			System.out.print("player: " + this.getPlayer() + " your possible moves are: ");
			for (int i = 0; i < moves.size(); i++) {
				System.out.print("(" +  moves.get(i).getRow() + ", " + moves.get(i).getColumn() + ") ");
			}
			System.out.print("");
			// checking the input and convert it to integers
			int row, column;
			boolean flag = true;
			do {
				System.out.print( "\nplease enter row number: ");
				Scanner console = new Scanner(System.in); 
				row = console.nextInt();
				System.out.print( "please enter column number: ");
				column = console.nextInt();
				Position p = new Position(row,column); // the position the player enter
				for (int i = 0; i < moves.size(); i++) {  // check that this position is equal to one of his options
					if (p.equal(moves.get(i))) {
						flag =false;
						break;
					}
				}
				if (flag){
					System.out.println("wrong input try again!! ");
				}
			} while (flag);
				Position p = new Position(row,column);
				logic.updateBoard(p,this.getPlayer());     // end of turn the main board is update and print
				logic.getBoard().print();
				return true;
		} else {
			return false;
		}
	}

}